extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceProducer.java

```diff
@@ -41,64 +40,72 @@
 import io.quarkus.runtime.configuration.ConfigurationException;
 import io.quarkus.vault.CredentialsProvider;
 
+/**
+ * This class is sort of a producer for {@link AgroalDataSource}.
+ *
+ * It isn't a CDI producer in the literal sense, but it is marked as a bean
+ * and it's {@code  createDataSource} method is called at runtime in order to produce
+ * the actual {@code AgroalDataSource} objects.
+ *
+ * CDI scopes and qualifiers are setup at build-time, which is why this class is devoid of
+ * any CDI annotations
+ *
+ */
 @SuppressWarnings("deprecation")
-public abstract class AbstractDataSourceProducer {
+public class DataSourceProducer {
 
-    private static final Logger log = Logger.getLogger(AbstractDataSourceProducer.class.getName());
+    private static final Logger log = Logger.getLogger(DataSourceProducer.class.getName());
 
-    private DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig;
-    private DataSourcesRuntimeConfig dataSourcesRuntimeConfig;
-    private DataSourcesJdbcBuildTimeConfig dataSourcesJdbcBuildTimeConfig;
-    private DataSourcesJdbcRuntimeConfig dataSourcesJdbcRuntimeConfig;
+    private final DataSourcesBuildTimeConfig dataSourcesBuildTimeConfig;
+    private final DataSourcesRuntimeConfig dataSourcesRuntimeConfig;
+    private final DataSourcesJdbcBuildTimeConfig dataSourcesJdbcBuildTimeConfig;
+    private final DataSourcesJdbcRuntimeConfig dataSourcesJdbcRuntimeConfig;
+    private final LegacyDataSourcesJdbcBuildTimeConfig legacyDataSourcesJdbcBuildTimeConfig;
+    private final LegacyDataSourcesRuntimeConfig legacyDataSourcesRuntimeConfig;
+    private final LegacyDataSourcesJdbcRuntimeConfig legacyDataSourcesJdbcRuntimeConfig;
```

Comment:
Can't we inject all that now if this is a CDI bean? I thought we could and tbh that was one of the reasons I pushed for this change :).

Reply:
We are injecting them, constructor injection :wink:

Reply:
Ah yeah, what misled me was I was expecting getters to be removed but we were not relying on getters anyway before.

Maybe we could drop `checkConfigInjection` then? Because we are sure things have been properly injected?

Reply:
Good idea, I fixed it